README.md

```diff
@@ -345,3 +346,136 @@ Types should never have prefixes because their names are already implicitly mang
+
+```swift
+class TableViewCell: UITableViewCell, ReusableView {
+    static let ReuseIdentifier: String = "TableViewCellIdentifier"
+    static let NibName: String = "CustomTableViewCell"
+}
+```
+> **Reasoning**: When registering cells for reuse in a UITableView or UICollectionView, you need the nib name to load the nib and the reuse identifier.
+
+#### NSNotification
+Name notifications in reverse domain format with the notification name in snake case.
+
+```swift
+com.linkedin.slideshare.notification_name
+```
+
+Create a struct to hold all notification names as constants.
+
+```swift
+struct GlobalNotifications {
+    static let ABC = ""
+}
+```
+
+Create notification handlers as lazy closures.
+
+```swift
+private lazy var handleNotificationABC: (NSNotification!) -> Void { [weak self] notification in
+    // Handle the notification
+}
+```
+> **Reasoning**: This way you can define capture semantics for self and also use the identifier as the selector in the addObserver method (see below) instead of a string. This gives you the safety of the compiler.
+
+Create a registerNotifications() method and deregisterNotifications().
+
+```swift
+func registerNotifications() {
+    let notificationCenter = NSNotificationCenter.defaultCenter()
+
+    notificationCenter.addObserver(self, selector: handleNotificationABC, name: GlobalNotifications.ABC, object: nil)
+}
+
+func deregisterNotifications() {
+    NSNotificationCenter.defaultCenter().removeObserver(self)
+}
+```
+
+#### View Controllers
+If the view controller is associated with a Storyboard, create a class method named createInstance to return an initialized instance of the view controller from the Storyboard.
+
+```swift
+static func createInstance() -> MasterViewController {
+    return UIStoryboard.initialControllerFromStoryboard("Master") as! MasterViewController
+}
+```
+> **Reasoning**: Use static if you are not going to make a subclass of this class. If you are going to make a subclass, change it to class func.
+
+If you have the situation described above, but have properties that need to be initialized, also create helper methods following the designated/convenience initializer type pattern like so.
+
+```swift
+static func createInstanceWithId(id: Int) -> MasterViewController {
+        let masterViewController = createInstance()
+        masterViewController.id = id
+        return masterViewController
+}
+```
+
+#### UIView
+If you have a class that inherits from UIView and has a XIB file where it is layed out, create a class method named createInstance similar to the example in the View Controllers section.
+
+```swift
+class CustomView: UIView {
+
+    private static let NibName: String = "CustomView"
+
+    static func createInstance() -> CustomView {
+        return NSBundle.mainBundle().loadNibNamed(nibName, owner: nil, options: nil)[0] as! CustomView
+    }
+}
+```

```

Comment:
Should we consider making this part of the protocol above as well?

Like 
protocol NibView {
var NibName:String { get }
}
protocol ReusableView : NibView {
var ReuseIdentifier:String {get} 
}